client/js/lounge.js

```diff
@@ -326,8 +335,16 @@ $(function() {
 	}
 
 	function renderChannelMessages(data) {
+		var channel = chat.find("#chan-" + data.id + " .messages");
 		var documentFragment = buildChannelMessages(data.id, data.messages);
-		var channel = chat.find("#chan-" + data.id + " .messages").append(documentFragment);
+
+		channel.append(documentFragment);
+
+		if (data.messages.length > 0) {
+			var lastSender = data.messages[data.messages.length - 1];
+			lastSender = lastSender.mode + lastSender.from;
+			channel.data("last-from", lastSender);
```

Comment:
Why this back-and-forth between storing in JS (`lastSender`) and storing in DOM (`data-last-from`)?

Isn't there a better way to do that, like looking at who the last message is from directly without storing that `data`?